lib/config/guiconfiguration.go

```diff
@@ -12,56 +12,74 @@ import (
 	"strings"
 )
 
-type GUIConfiguration struct {
-	Enabled                   bool   `xml:"enabled,attr" json:"enabled" default:"true"`
-	RawAddress                string `xml:"address" json:"address" default:"127.0.0.1:8384"`
-	User                      string `xml:"user,omitempty" json:"user"`
-	Password                  string `xml:"password,omitempty" json:"password"`
-	RawUseTLS                 bool   `xml:"tls,attr" json:"useTLS"`
-	APIKey                    string `xml:"apikey,omitempty" json:"apiKey"`
+type GUIListener struct {
+	Address                   string `xml:"address" json:"address"`
+	UseTLS                    bool   `xml:"tls,attr" json:"useTLS"`
 	InsecureAdminAccess       bool   `xml:"insecureAdminAccess,omitempty" json:"insecureAdminAccess"`
-	Theme                     string `xml:"theme" json:"theme" default:"default"`
-	Debugging                 bool   `xml:"debugging,attr" json:"debugging"`
 	InsecureSkipHostCheck     bool   `xml:"insecureSkipHostcheck,omitempty" json:"insecureSkipHostcheck"`
 	InsecureAllowFrameLoading bool   `xml:"insecureAllowFrameLoading,omitempty" json:"insecureAllowFrameLoading"`
 }
 
-func (c GUIConfiguration) Address() string {
-	if override := os.Getenv("STGUIADDRESS"); override != "" {
-		// This value may be of the form "scheme://address:port" or just
-		// "address:port". We need to chop off the scheme. We try to parse it as
-		// an URL if it contains a slash. If that fails, return it as is and let
-		// some other error handling handle it.
-
-		if strings.Contains(override, "/") {
-			url, err := url.Parse(override)
-			if err != nil {
-				return override
-			}
-			return url.Host
-		}
+type GUIConfiguration struct {
+	Listeners []GUIListener `xml:"listener" json:"listeners"`
+	Enabled   bool          `xml:"enabled,attr" json:"enabled" default:"true"`
+	User      string        `xml:"user,omitempty" json:"user"`
+	Password  string        `xml:"password,omitempty" json:"password"`
+	APIKey    string        `xml:"apikey,omitempty" json:"apiKey"`
+	Theme     string        `xml:"theme" json:"theme" default:"default"`
+	Debugging bool          `xml:"debugging,attr" json:"debugging"`
 
-		return override
-	}
+	// Deprecated. Old listener configuration style.
+	Deprecated_RawAddress                string `xml:"address,omitempty" json:"-"`
+	Deprecated_RawUseTLS                 bool   `xml:"tls,attr,omitempty" json:"-"`
+	Deprecated_InsecureAdminAccess       bool   `xml:"insecureAdminAccess,omitempty" json:"-"`
+	Deprecated_InsecureSkipHostCheck     bool   `xml:"insecureSkipHostcheck,omitempty" json:"-"`
+	Deprecated_InsecureAllowFrameLoading bool   `xml:"insecureAllowFrameLoading,omitempty" json:"-"`
+}
 
-	return c.RawAddress
+func GUIListenerFromEnv(envAddr string) (l GUIListener) {
+	if strings.Contains(envAddr, "/") {
+		url, err := url.Parse(envAddr)
+		if err != nil {
+			l.Address = envAddr
+		} else {
+			l.Address = url.Host
+		}
+	}
```

Comment:
This will end up with empty address in case of `STGUIADDRESSES=127.0.0.1:1234`

Reply:
Right!